cvat-ui/src/actions/auth-actions.ts

```diff
@@ -52,11 +52,10 @@ export const registerAsync = (
     dispatch(authActions.register());
 
     try {
-        await cvat.server.register(username, firstName, lastName, email, password1, password2,
+        const user = await cvat.server.register(username, firstName, lastName, email, password1, password2,
             confirmations);
-        const users = await cvat.users.get({ self: true });
 
-        dispatch(authActions.registerSuccess(users[0]));
+        dispatch(authActions.registerSuccess(user));
```

Comment:
users[0] was an instance of the class ``User``
But register method returns just an ``Object``. This object has a bit different fields, you can look at user constructor: ``cvat-core/src/user.js``

Reply:
Also I can suggest to add field: ``isVerified = !email_verification_required;`` to this class and use it anywhere instead of ``email_verification_required``